jax/interpreters/pxla.py

```diff
@@ -761,6 +778,25 @@ def dynamic_fun(dummy, *args):
 multi_host_supported_collectives: Set[core.Primitive] = set()
 
 
+PartitionsOrReplicated = Optional[Tuple[int, ...]]
+
+def _find_partitions(jaxpr) -> Tuple[
+    Optional[Tuple[PartitionsOrReplicated, ...]],
+    Optional[Tuple[PartitionsOrReplicated, ...]],
+    int]:
+  """Returns (in_partitions, out_partitions, num_partitions)."""
+  for eqn in jaxpr.eqns:
+    if eqn.primitive.name == "sharded_call":
```

Comment:
Is there a need to look for `sharded_call` deeper in the jaxpr?

Reply:
Not yet, because we don't currently support pmap of sharded_jit + non-sharded operations.

Reply:
It would be good to raise an error in this situation though. This is currently a sharp edge in general with `sharded_jit`, since it doesn't work under other transformations (e.g. jit), but it won't stop you from trying. I'd like to improve this situation, either by figuring out how composition should work or at least figuring out how to raise good errors, but not in this change :)